stdlib/public/core/ArrayBuffer.swift

```diff
@@ -567,6 +571,78 @@ extension _ArrayBuffer {
     }
   }
 
+  @inlinable @_alwaysEmitIntoClient
+  static var associationKey: UnsafeRawPointer {
+    //We never dereference this, we just need an address to use as a unique key
+    UnsafeRawPointer(Builtin.addressof(&_swiftEmptyArrayStorage))
+  }
+  
+  @inlinable @_alwaysEmitIntoClient
+  internal func getAssociatedBuffer() -> _ContiguousArrayBuffer<Element>? {
+    let getter = objc_getAssociatedObject as @convention(c)(
```

Comment:
See comment below about objc_setAssociatedObject for what's going on here. It's almost the same trick but slightly less tangly because we don't have to deal with objc_associationPolicy